helix-view/src/keyboard.rs

```diff
@@ -53,6 +53,164 @@ impl From<crossterm::event::KeyModifiers> for KeyModifiers {
+pub enum MediaKeyCode {
+    /// Play media key.
+    Play,
+    /// Pause media key.
+    Pause,
+    /// Play/Pause media key.
+    PlayPause,
+    /// Reverse media key.
+    Reverse,
+    /// Stop media key.
+    Stop,
+    /// Fast-forward media key.
+    FastForward,
+    /// Rewind media key.
+    Rewind,
+    /// Next-track media key.
+    TrackNext,
+    /// Previous-track media key.
+    TrackPrevious,
+    /// Record media key.
+    Record,
+    /// Lower-volume media key.
+    LowerVolume,
+    /// Raise-volume media key.
+    RaiseVolume,
+    /// Mute media key.
+    MuteVolume,
+}
+
+#[cfg(feature = "term")]
+impl From<MediaKeyCode> for crossterm::event::MediaKeyCode {
+    fn from(media_key_code: MediaKeyCode) -> Self {
+        use crossterm::event::MediaKeyCode as CMediaKeyCode;
+
+        match media_key_code {
+            MediaKeyCode::Play => CMediaKeyCode::Play,
+            MediaKeyCode::Pause => CMediaKeyCode::Pause,
+            MediaKeyCode::PlayPause => CMediaKeyCode::PlayPause,
+            MediaKeyCode::Reverse => CMediaKeyCode::Reverse,
+            MediaKeyCode::Stop => CMediaKeyCode::Stop,
+            MediaKeyCode::FastForward => CMediaKeyCode::FastForward,
+            MediaKeyCode::Rewind => CMediaKeyCode::Rewind,
+            MediaKeyCode::TrackNext => CMediaKeyCode::TrackNext,
+            MediaKeyCode::TrackPrevious => CMediaKeyCode::TrackPrevious,
+            MediaKeyCode::Record => CMediaKeyCode::Record,
+            MediaKeyCode::LowerVolume => CMediaKeyCode::LowerVolume,
+            MediaKeyCode::RaiseVolume => CMediaKeyCode::RaiseVolume,
+            MediaKeyCode::MuteVolume => CMediaKeyCode::MuteVolume,
+        }
+    }
+}
+
+#[cfg(feature = "term")]
+impl From<crossterm::event::MediaKeyCode> for MediaKeyCode {
+    fn from(val: crossterm::event::MediaKeyCode) -> Self {
+        use crossterm::event::MediaKeyCode as CMediaKeyCode;
+
+        match val {
+            CMediaKeyCode::Play => MediaKeyCode::Play,
+            CMediaKeyCode::Pause => MediaKeyCode::Pause,
+            CMediaKeyCode::PlayPause => MediaKeyCode::PlayPause,
+            CMediaKeyCode::Reverse => MediaKeyCode::Reverse,
+            CMediaKeyCode::Stop => MediaKeyCode::Stop,
+            CMediaKeyCode::FastForward => MediaKeyCode::FastForward,
+            CMediaKeyCode::Rewind => MediaKeyCode::Rewind,
+            CMediaKeyCode::TrackNext => MediaKeyCode::TrackNext,
+            CMediaKeyCode::TrackPrevious => MediaKeyCode::TrackPrevious,
+            CMediaKeyCode::Record => MediaKeyCode::Record,
+            CMediaKeyCode::LowerVolume => MediaKeyCode::LowerVolume,
+            CMediaKeyCode::RaiseVolume => MediaKeyCode::RaiseVolume,
+            CMediaKeyCode::MuteVolume => MediaKeyCode::MuteVolume,
+        }
+    }
+}
+
+/// Represents a media key (as part of [`KeyCode::Modifier`]).
+#[derive(Debug, PartialOrd, Ord, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum ModifierKeyCode {
```

Comment:
Modifier keys are controlled with a different flag (see [crossterm's event-read example](https://github.com/crossterm-rs/crossterm/blob/b13e8ef64d57a3b9a642194b6beb069b55d6ef43/examples/event-read.rs#L77-L81)) and we haven't seen reports of terminals sending these without the flag, but I have added them here for completeness.